test/page.spec.js

```diff
@@ -781,6 +781,126 @@ module.exports.addTests = function({testRunner, expect, puppeteer, DeviceDescrip
     });
   });
 
+  describe('Page.waitForRequest', function() {
+    it('should work', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+
+      server.setRoute('/getRequest', (req, res) => {
+        res.statusCode = 200;
+        res.end();
+      });
+
+      page.evaluate(() => {
+        fetch(`/getRequest`, { method: 'GET'});
+        Promise.resolve();
+      });
+
+      const request = await page.waitForRequest(`${server.PREFIX}/getRequest`);
+
+      expect(request._method).toBe('GET');
+      expect(request.url()).toBe(`${server.PREFIX}/getRequest`);
+    });
+
+    it('should work with regex pattern', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+
+      server.setRoute('/getRequest', (req, res) => {
+        res.statusCode = 200;
+        res.end();
+      });
+
+      page.evaluate(() => {
+        fetch(`/getRequest`, { method: 'GET'});
+        Promise.resolve();
+      });
+
+      const request = await page.waitForRequest(new RegExp(`${server.PREFIX}/(get|test)Request`));
+
+      expect(request._method).toBe('GET');
+      expect(request.url()).toBe(`${server.PREFIX}/getRequest`);
+    });
+
+    it('should work with method option', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+
+      server.setRoute('/getRequest', (req, res) => {
+        res.statusCode = 200;
+        res.end();
+      });
+
+      page.evaluate(() => {
+        fetch(`/getRequest`, { method: 'PATCH'});
+        Promise.resolve();
+      });
+
+      const request = await page.waitForRequest(`${server.PREFIX}/getRequest`, { method: ['POST', 'PATCH']});
+
+      expect(request._method).toBe('PATCH');
+      expect(request.url()).toBe(`${server.PREFIX}/getRequest`);
+    });
+  });
+
+  describe('Page.waitForResponse', function() {
+    it('should work', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+
+      server.setRoute('/testResponse', (req, res) => serverResponse = () => res.end('Hello'));
+
+      page.evaluate(() => fetch(`/testResponse`, { method: 'GET'}).then(() => true));
+      setTimeout(() => serverResponse(), 100);
+
+      const response = await page.waitForResponse(`${server.PREFIX}/testResponse`);
+
+      expect(response.ok()).toBe(true);
+      expect(response.url()).toBe(`${server.PREFIX}/testResponse`);
+    });
+
+    it('should work with regex', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+
+      server.setRoute('/testResponse', (req, res) => serverResponse = () => res.end('Hello'));
```

Comment:
you can avoid setting routes if you just query for some of the resources in the `test/assets/` folder